fgpyo/sam/__init__.py

```diff
@@ -1221,6 +1288,31 @@ def __next__(self) -> Template:
         return Template.build(recs, validate=False)
 
 
+def set_mate_info_for_template(
```

Comment:
I think I'd prefer this to be an instance method on `Template`, what do you think?

Reply:
Currently I have it as both. But since the function always requires a template it doesn't make sense to have both forms of the same function since they don't vary in any other way. I'll add this only to the template for now. If someone wants a standalone function, they can make an argument and refactor later.